src/uct/ib/ud/base/ud_iface.c

```diff
@@ -371,6 +372,40 @@ void uct_ud_iface_remove_async_handlers(uct_ud_iface_t *iface)
     ucs_async_remove_handler(iface->async.timer_id, 1);
 }
 
+/* Calculate real GIDs len. Can be either 16 (RoCEv1 or RoCEv2/IPv6)
+ * or 4 (RoCEv2/IPv4). This len is used for packets filtering by DGIDs.
+ *
+ * According to Annex17_RoCEv2 (A17.4.5.2):
+ * "The first 40 bytes of user posted UD Receive Buffers are reserved for the L3
+ * header of the incoming packet (as per the InfiniBand Spec Section 11.4.1.2).
+ * In RoCEv2, this area is filled up with the IP header. IPv6 header uses the
+ * entire 40 bytes. IPv4 headers use the 20 bytes in the second half of the
+ * reserved 40 bytes area (i.e. offset 20 from the beginning of the receive
+ * buffer). In this case, the content of the first 20 bytes is undefined." */
+static void uct_ud_iface_parse_gid_format(uct_ud_iface_t *iface)
+{
+    const int ipv4_len = 4;
+    const int ipv6_len = 16;
+    uint16_t *first    = (uint16_t*)iface->super.gid.raw;
```

Comment:
instead of defining these, i would do
uint16_t *dgid_u16 = (uint16_t*)iface->super.gid.raw;
and then check dgid_u16[0] (aka "first") and dgid_u16[5] (aka "last")